sdk/cosmos/src/client_builder.rs

```diff
@@ -0,0 +1,83 @@
+use crate::clients::{
+    ChinaCosmosUri, ClientBuilder, CustomCosmosUri, DefaultCosmosUri, HTTPClientNotAssigned,
+};
+use crate::AuthorizationToken;
+use azure_core::errors::AzureError;
+use std::borrow::Cow;
+use std::marker::PhantomData;
+
+pub fn new<'a, IntoCowStr>(
```

Comment:
Agreed, a free floating `new` is ugly and confusing. It was inside a struct before, I have moved it but haven't changed the names. :bowing_man: 

What about `build_default_client`, `build_china_client`, `build_custom_client`, `build_emulator_client`? 

The usage will become something like:

```rust
   let client = a<zure_cosmos::client_builder::build_default_client(&account, authorization_token)?
```

instead of: 

```rust
   let client = azure_cosmos::client_builder::new(&account, authorization_token)?
```

But if the user imports the function directly the former becomes much more telling:

```rust
   let client = build_default_client(&account, authorization_token)?
```

instead of 

```rust
   // this is ugly
   let client = new(&account, authorization_token)?
```